python/ray/worker.py

```diff
@@ -1536,36 +1470,6 @@ def connect(node,
             worker.logger_thread.daemon = True
             worker.logger_thread.start()
 
-    if mode == SCRIPT_MODE:
-        # Add the directory containing the script that is running to the Python
-        # paths of the workers. Also add the current directory. Note that this
-        # assumes that the directory structures on the machines in the clusters
-        # are the same.
-        # When using an interactive shell, there is no script directory.
-        if not interactive_mode:
-            script_directory = os.path.abspath(os.path.dirname(sys.argv[0]))
-            worker.run_function_on_all_workers(
-                lambda worker_info: sys.path.insert(1, script_directory))
-        # In client mode, if we use runtime envs with "working_dir", then
-        # it'll be handled automatically.  Otherwise, add the current dir.
-        if not job_config.client_job and not job_config.runtime_env_has_uris():
-            current_directory = os.path.abspath(os.path.curdir)
-            worker.run_function_on_all_workers(
-                lambda worker_info: sys.path.insert(1, current_directory))
```

Comment:
@ericl @architkulkarni I think we should be safe to remove this given that we have runtime_env now, but I'm not sure if it'll break some applications that have a weird assumption on this behavior.

For example, if someone was rsync-ing their directory to the whole cluster and then attempting to import from files that were in that directory, I think it might work as-is due to this code and would break without it.

Thoughts? I'm honestly not that sure what the context of why we have this logic is.

Reply:
Iirc that's exactly the use case this covers (@iycheng )